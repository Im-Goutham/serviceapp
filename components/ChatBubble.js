/* eslint-disable no-underscore-dangle, no-use-before-define */

import PropTypes from 'prop-types';
import React from 'react';
import {
  Text,
  Clipboard,
  StyleSheet,
  TouchableOpacity,
  View,
  ViewPropTypes,
  Platform,
  Image
} from 'react-native';

import { MessageText, MessageImage, Time, utils } from 'react-native-gifted-chat';
import LinearGradient from 'react-native-linear-gradient';

const { isSameUser, isSameDay } = utils;

export default class Bubble extends React.Component {

  constructor(props) {
    super(props);
    this.onLongPress = this.onLongPress.bind(this);
  }

  onLongPress() {
    if (this.props.onLongPress) {
      this.props.onLongPress(this.context);
    } else if (this.props.currentMessage.text) {
      const options = [
        'Copy Text',
        'Cancel',
      ];
      const cancelButtonIndex = options.length - 1;
      this.context.actionSheet().showActionSheetWithOptions(
        { options, cancelButtonIndex },
        (buttonIndex) => {
          if (buttonIndex === 0) {
            Clipboard.setString(this.props.currentMessage.text);
          }
        });
    }
  }

  renderMessageText() {
    if (this.props.currentMessage.text) {
      const { containerStyle, wrapperStyle, messageTextStyle, ...messageTextProps } = this.props;
      if (this.props.renderMessageText) {
        return this.props.renderMessageText(messageTextProps);
      }
      return (
        <MessageText
        {...messageTextProps}
        textStyle={{
          left: [styles.standardFont, styles.slackMessageText, messageTextProps.textStyle, messageTextStyle],
        }}
      />
      );
    }
    return null;
  }

  renderMessageImage() {
    if (this.props.currentMessage.image) {
      const { containerStyle, wrapperStyle, ...messageImageProps } = this.props;
      if (this.props.renderMessageImage) {
        return this.props.renderMessageImage(messageImageProps);
      }
      return <MessageImage {...messageImageProps} imageStyle={[styles.slackImage, messageImageProps.imageStyle]} />;
    }
    return null;
  }

  renderTicks() {
    const { currentMessage } = this.props;
    if (this.props.renderTicks) {
      return this.props.renderTicks(currentMessage);
    }
    if (currentMessage.user._id !== this.props.user._id) {
      return null;
    }
    if (currentMessage.sent || currentMessage.received) {
      return (
        <View style={[styles.headerItem, styles.tickView]}>
          {currentMessage.sent && <Text style={[styles.standardFont, styles.tick, this.props.tickStyle]}>✓</Text>}
          {currentMessage.received && <Text style={[styles.standardFont, styles.tick, this.props.tickStyle]}>✓</Text>}
        </View>
      );
    }
    return null;
  }

  renderUsername() {
    const username = this.props.currentMessage.user.name;
    if (username) {
      const { containerStyle, wrapperStyle, ...usernameProps } = this.props;
      if (this.props.renderUsername) {
        return this.props.renderUsername(usernameProps);
      }
      return (
        <Text style={[styles.standardFont, styles.headerItem, styles.username, this.props.usernameStyle]}>
          {username}
        </Text>
      );
    }
    return null;
  }

  renderTime() {
    if (this.props.currentMessage.createdAt) {
      const { containerStyle, wrapperStyle, ...timeProps } = this.props;
      if (this.props.renderTime) {
        return this.props.renderTime(timeProps);
      }
      return (
        <Time
          {...timeProps}
          containerStyle={{ left: [styles.timeContainer] }}
          textStyle={{ left: [styles.standardFont, styles.headerItem, styles.time, timeProps.textStyle] }}
        />
      );
    }
    return null;
  }

  renderCustomView() {
    if (this.props.renderCustomView) {
      return this.props.renderCustomView(this.props);
    }
    return null;
  }

  render() {
    const isSameThread = isSameUser(this.props.currentMessage, this.props.previousMessage)
      && isSameDay(this.props.currentMessage, this.props.previousMessage);

    const messageHeader = isSameThread ? null : (
      <View style={styles.headerView}>
        {this.renderUsername()}
        {this.renderTime()}
        {this.renderTicks()}
      </View>
    );
   console.log("this.props.currentMessage are" ,this.props.currentMessage);
   var gradiant = ['#3EBDEF','#3E85EF'];
   if(this.props.currentMessage.sent){
      gradiant = ['#EAF8FF','#C6E1FF'];
   }
   let {text, sent } = this.props.currentMessage;
   if(!sent){
      return (
        <View style={{flex:1,flexDirection:'row',paddingHorizontal:10}}>
             <View style={styles.imgBox}>
             <Image source={require('../images/svp1.png')} style={{borderRadius:25,width:50,height:50}}/>
            </View>
            <View style={styles.textBox}>
            <LinearGradient
              colors={gradiant}
              start={{x: 0, y: 0}}
              end={{x: 1, y: 0}}
              style={[styles.gradiantStyle,{borderBottomLeftRadius:0}]}>
              <Text style={[styles.textStyle]}>{text}</Text>
              <Text style={[styles.dateTextStyle]}>2.23 PM</Text>
          </LinearGradient>
            </View>
        </View>

      );
   }
   else {
    return (
     <View style={{flex:1,flexDirection:'row',paddingLeft:10,paddingRight:10}}>
          <View style={{flex:8}}>
            <LinearGradient
              colors={gradiant}
              start={{x: 0, y: 0}}
              end={{x: 1, y: 0}}
              style={[styles.gradiantStyle,{borderBottomRightRadius:0}]}>
              <Text style={[styles.textStyle,{color:'black'}]}>{text}</Text>
              <Text style={[styles.dateTextStyle,{color:'#rgb(155,155,155)'}]}>2.23 PM</Text>
          </LinearGradient>
          </View>
          <View style={styles.imgBox}>
           <Image source={require('../images/svp2.png')} style={{borderRadius:25,width:50,height:50}}/>
          </View>
      </View>

    );
   }
 
  }

}

// Note: Everything is forced to be "left" positioned with this component.
// The "right" position is only used in the default Bubble.
const styles = StyleSheet.create({
  standardFont: {
    fontSize: 15,
  },
  imgBox: {
    flex:2,
    justifyContent:'flex-end',
    alignItems:'center'
  },
  textBox: {
    flex:8,
    borderRadius:10,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 2,
    elevation: 3,
},
gradiantStyle: {
  padding :20,
  borderRadius:10
},
textStyle:{
  textAlign:'left',
  fontSize:16,
  fontFamily:'Montserrat-Regular',
  color:'white'
},
dateTextStyle:{
  fontFamily:'Montserrat-Bold',
  textAlign:'right',
  color:'white',
  paddingTop:8
},
  slackMessageText: {
    marginLeft: 0,
    marginRight: 0,
  },
  container: {
    flex: 1,
    backgroundColor:'red',
    alignItems: 'flex-start',
  },
  wrapper: {
    marginRight: 60,
    minHeight: 20,
    justifyContent: 'flex-end',
  },
  username: {
    fontWeight: 'bold',
  },
  time: {
    textAlign: 'left',
    fontSize: 12,
  },
  timeContainer: {
    marginLeft: 0,
    marginRight: 0,
    marginBottom: 0,
  },
  headerItem: {
    marginRight: 10,
  },
  headerView: {
    // Try to align it better with the avatar on Android.
    marginTop: Platform.OS === 'android' ? -2 : 0,
    flexDirection: 'row',
    alignItems: 'baseline',
  },
  /* eslint-disable react-native/no-color-literals */
  tick: {
    backgroundColor: 'transparent',
    color: 'white',
  },
  /* eslint-enable react-native/no-color-literals */
  tickView: {
    flexDirection: 'row',
  },
  slackImage: {
    borderRadius: 3,
    marginLeft: 0,
    marginRight: 0,
  },
});

Bubble.contextTypes = {
  actionSheet: PropTypes.func,
};

Bubble.defaultProps = {
  touchableProps: {},
  onLongPress: null,
  renderMessageImage: null,
  renderMessageText: null,
  renderCustomView: null,
  renderTime: null,
  currentMessage: {
    text: null,
    createdAt: null,
    image: null,
  },
  nextMessage: {},
  previousMessage: {},
  containerStyle: {},
  wrapperStyle: {},
  tickStyle: {},
  containerToNextStyle: {},
  containerToPreviousStyle: {},
};

Bubble.propTypes = {
  touchableProps: PropTypes.object,
  onLongPress: PropTypes.func,
  renderMessageImage: PropTypes.func,
  renderMessageText: PropTypes.func,
  renderCustomView: PropTypes.func,
  renderUsername: PropTypes.func,
  renderTime: PropTypes.func,
  renderTicks: PropTypes.func,
  currentMessage: PropTypes.object,
  nextMessage: PropTypes.object,
  previousMessage: PropTypes.object,
  user: PropTypes.object,
  containerStyle: PropTypes.shape({
    left: ViewPropTypes.style,
    right: ViewPropTypes.style,
  }),
  wrapperStyle: PropTypes.shape({
    left: ViewPropTypes.style,
    right: ViewPropTypes.style,
  }),
  messageTextStyle: Text.propTypes.style,
  usernameStyle: Text.propTypes.style,
  tickStyle: Text.propTypes.style,
  containerToNextStyle: PropTypes.shape({
    left: ViewPropTypes.style,
    right: ViewPropTypes.style,
  }),
  containerToPreviousStyle: PropTypes.shape({
    left: ViewPropTypes.style,
    right: ViewPropTypes.style,
  }),
};