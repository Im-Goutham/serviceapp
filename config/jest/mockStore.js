export const initialState = {
    user: {
      user: {},
      backButton: false
    },
    photo: {
      photoView: false,
      images: [],
      index: 0
    }
  }