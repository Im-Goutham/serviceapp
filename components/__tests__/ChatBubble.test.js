jest.unmock('redux-mock-store')
jest.unmock('redux-thunk')
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';
import ChatBubble from '../ChatBubble';
import {initialState} from '../../config/jest/mockStore';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let store = mockStore(initialState);


describe('ChatBubble', () => {
    let component;
    const mockFunc = jest.fn();
    const navigation = {
        navigate: mockFunc,
        goBack: mockFunc
    }
    
    const currentMessage = {
         user: {name : ''}
     };

     const user = {
          id:''
     }

    beforeEach(() => { 
    component = shallow(<ChatBubble store={store} onPress={mockFunc} navigation={navigation} currentMessage={currentMessage} user={user}/>).dive();
    })

    describe('Snapshot testing',()=>{
        it('should have same snapshot',()=> {
            expect(component).toMatchSnapshot();
        })
     })

     describe('Function testing',()=>{
    
     })


    //   describe('Form field testing',()=>{
           
    //         it('should press signUp button ',()=>{
    //                 component.find({testID:'signUpButton'}).simulate('press');
    //                 expect(mockFunc).toHaveBeenCalled();
    //         })
    //     })
});
