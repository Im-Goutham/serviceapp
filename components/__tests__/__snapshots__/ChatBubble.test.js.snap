// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBubble Snapshot testing should have same snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingHorizontal": 8.695652173913045,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "flex-end",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/svp1.png",
          }
        }
        style={
          Object {
            "borderRadius": 21.73913043478261,
            "height": 43.47826086956522,
            "width": 43.47826086956522,
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "borderRadius": 8.695652173913045,
          "elevation": 3,
          "flex": 8,
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2,
        }
      }
    >
      <LinearGradient
        colors={
          Array [
            "#3EBDEF",
            "#3E85EF",
          ]
        }
        end={
          Object {
            "x": 1,
            "y": 0,
          }
        }
        start={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 8.695652173913045,
              "padding": 17.39130434782609,
            },
            Object {
              "borderBottomLeftRadius": 0,
            },
          ]
        }
      >
        <Component
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 13.913043478260871,
                "textAlign": "left",
              },
            ]
          }
        />
        <Component
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 10.434782608695652,
                "paddingTop": 6.9565217391304355,
                "textAlign": "right",
              },
            ]
          }
        >
          2.23 PM
        </Component>
      </LinearGradient>
    </Component>
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "justifyContent": "flex-end",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../images/svp1.png",
              }
            }
            style={
              Object {
                "borderRadius": 21.73913043478261,
                "height": 43.47826086956522,
                "width": 43.47826086956522,
              }
            }
          />
        </Component>,
        <Component
          style={
            Object {
              "borderRadius": 8.695652173913045,
              "elevation": 3,
              "flex": 8,
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 2,
            }
          }
        >
          <LinearGradient
            colors={
              Array [
                "#3EBDEF",
                "#3E85EF",
              ]
            }
            end={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            start={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 8.695652173913045,
                  "padding": 17.39130434782609,
                },
                Object {
                  "borderBottomLeftRadius": 0,
                },
              ]
            }
          >
            <Component
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Regular",
                    "fontSize": 13.913043478260871,
                    "textAlign": "left",
                  },
                ]
              }
            />
            <Component
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 10.434782608695652,
                    "paddingTop": 6.9565217391304355,
                    "textAlign": "right",
                  },
                ]
              }
            >
              2.23 PM
            </Component>
          </LinearGradient>
        </Component>,
      ],
      "style": Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingHorizontal": 8.695652173913045,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Image
            source={
              Object {
                "testUri": "../../../images/svp1.png",
              }
            }
            style={
              Object {
                "borderRadius": 21.73913043478261,
                "height": 43.47826086956522,
                "width": 43.47826086956522,
              }
            }
          />,
          "style": Object {
            "alignItems": "center",
            "flex": 2,
            "justifyContent": "flex-end",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": Object {
              "testUri": "../../../images/svp1.png",
            },
            "style": Object {
              "borderRadius": 21.73913043478261,
              "height": 43.47826086956522,
              "width": 43.47826086956522,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <LinearGradient
            colors={
              Array [
                "#3EBDEF",
                "#3E85EF",
              ]
            }
            end={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            start={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 8.695652173913045,
                  "padding": 17.39130434782609,
                },
                Object {
                  "borderBottomLeftRadius": 0,
                },
              ]
            }
          >
            <Component
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Regular",
                    "fontSize": 13.913043478260871,
                    "textAlign": "left",
                  },
                ]
              }
            />
            <Component
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 10.434782608695652,
                    "paddingTop": 6.9565217391304355,
                    "textAlign": "right",
                  },
                ]
              }
            >
              2.23 PM
            </Component>
          </LinearGradient>,
          "style": Object {
            "borderRadius": 8.695652173913045,
            "elevation": 3,
            "flex": 8,
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 2,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Component
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 13.913043478260871,
                      "textAlign": "left",
                    },
                  ]
                }
              />,
              <Component
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Bold",
                      "fontSize": 10.434782608695652,
                      "paddingTop": 6.9565217391304355,
                      "textAlign": "right",
                    },
                  ]
                }
              >
                2.23 PM
              </Component>,
            ],
            "colors": Array [
              "#3EBDEF",
              "#3E85EF",
            ],
            "end": Object {
              "x": 1,
              "y": 0,
            },
            "start": Object {
              "x": 0,
              "y": 0,
            },
            "style": Array [
              Object {
                "borderRadius": 8.695652173913045,
                "padding": 17.39130434782609,
              },
              Object {
                "borderBottomLeftRadius": 0,
              },
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "style": Array [
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Regular",
                    "fontSize": 13.913043478260871,
                    "textAlign": "left",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "2.23 PM",
                "style": Array [
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 10.434782608695652,
                    "paddingTop": 6.9565217391304355,
                    "textAlign": "right",
                  },
                ],
              },
              "ref": null,
              "rendered": "2.23 PM",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "alignItems": "center",
                "flex": 2,
                "justifyContent": "flex-end",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../images/svp1.png",
                }
              }
              style={
                Object {
                  "borderRadius": 21.73913043478261,
                  "height": 43.47826086956522,
                  "width": 43.47826086956522,
                }
              }
            />
          </Component>,
          <Component
            style={
              Object {
                "borderRadius": 8.695652173913045,
                "elevation": 3,
                "flex": 8,
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 2,
              }
            }
          >
            <LinearGradient
              colors={
                Array [
                  "#3EBDEF",
                  "#3E85EF",
                ]
              }
              end={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              start={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 8.695652173913045,
                    "padding": 17.39130434782609,
                  },
                  Object {
                    "borderBottomLeftRadius": 0,
                  },
                ]
              }
            >
              <Component
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 13.913043478260871,
                      "textAlign": "left",
                    },
                  ]
                }
              />
              <Component
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Bold",
                      "fontSize": 10.434782608695652,
                      "paddingTop": 6.9565217391304355,
                      "textAlign": "right",
                    },
                  ]
                }
              >
                2.23 PM
              </Component>
            </LinearGradient>
          </Component>,
        ],
        "style": Object {
          "flex": 1,
          "flexDirection": "row",
          "paddingHorizontal": 8.695652173913045,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              source={
                Object {
                  "testUri": "../../../images/svp1.png",
                }
              }
              style={
                Object {
                  "borderRadius": 21.73913043478261,
                  "height": 43.47826086956522,
                  "width": 43.47826086956522,
                }
              }
            />,
            "style": Object {
              "alignItems": "center",
              "flex": 2,
              "justifyContent": "flex-end",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": Object {
                "testUri": "../../../images/svp1.png",
              },
              "style": Object {
                "borderRadius": 21.73913043478261,
                "height": 43.47826086956522,
                "width": 43.47826086956522,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <LinearGradient
              colors={
                Array [
                  "#3EBDEF",
                  "#3E85EF",
                ]
              }
              end={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              start={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 8.695652173913045,
                    "padding": 17.39130434782609,
                  },
                  Object {
                    "borderBottomLeftRadius": 0,
                  },
                ]
              }
            >
              <Component
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 13.913043478260871,
                      "textAlign": "left",
                    },
                  ]
                }
              />
              <Component
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Bold",
                      "fontSize": 10.434782608695652,
                      "paddingTop": 6.9565217391304355,
                      "textAlign": "right",
                    },
                  ]
                }
              >
                2.23 PM
              </Component>
            </LinearGradient>,
            "style": Object {
              "borderRadius": 8.695652173913045,
              "elevation": 3,
              "flex": 8,
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 2,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Component
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontFamily": "Montserrat-Regular",
                        "fontSize": 13.913043478260871,
                        "textAlign": "left",
                      },
                    ]
                  }
                />,
                <Component
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontFamily": "Montserrat-Bold",
                        "fontSize": 10.434782608695652,
                        "paddingTop": 6.9565217391304355,
                        "textAlign": "right",
                      },
                    ]
                  }
                >
                  2.23 PM
                </Component>,
              ],
              "colors": Array [
                "#3EBDEF",
                "#3E85EF",
              ],
              "end": Object {
                "x": 1,
                "y": 0,
              },
              "start": Object {
                "x": 0,
                "y": 0,
              },
              "style": Array [
                Object {
                  "borderRadius": 8.695652173913045,
                  "padding": 17.39130434782609,
                },
                Object {
                  "borderBottomLeftRadius": 0,
                },
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": undefined,
                  "style": Array [
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 13.913043478260871,
                      "textAlign": "left",
                    },
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "2.23 PM",
                  "style": Array [
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Bold",
                      "fontSize": 10.434782608695652,
                      "paddingTop": 6.9565217391304355,
                      "textAlign": "right",
                    },
                  ],
                },
                "ref": null,
                "rendered": "2.23 PM",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
